File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.java
import java.util.Arrays;

class Solution {
	public int solution(int[] A, int[] B) {
		int i;
		int idx;

		Arrays.sort(A);
		Arrays.sort(B);
		idx = B.length - 1;
		for (i = A.length - 1; i >= 0; i--) {
			if (A[i] < B[idx]) {
				idx--;
			}
		}
		return B.length - 1 - idx;
	}
}

File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> A, vector<int> B) {
    int i;
    int j;
    int sum;
    int size;

    sort(A.begin(), A.end());
    sort(B.begin(), B.end());
    size = A.size();
    sum = 0;
    for (i = 0, j = -1; i < size; i++) {
        for (++j; j < size && A[i] >= B[j]; j++);
        if (j == size) {
            break;
        }
    }
    return i;
}